test/saveHashTable.m

```diff
@@ -1,37 +1,159 @@
+                    % in the wrong environment for the test.
+                end
+            else
+                warning('SaveHashTable:NoHashFound',...
+                        'No hash found for "%s"!', thisFunc);
+            end
+            if ~isempty(thisHash)
+                hashes.(thisFunc) = thisHash;
+            end
+        end
+
+        % ask what to do with tests for which we have a hash, but no test results
+        removedTests = setdiff(oldFunctions, newFunctions);
+        if ~isempty(removedTests)
+            fprintf(1, 'Some tests in the file were not in the build status.\n');
+        end
+        for iTest = 1:numel(removedTests)
+            thisTest = removedTests{iTest};
+
+            action = askActionToPerformOnRemovedTest(thisTest);
+            switch action
+                case 'remove'
+                    % useful for test that no longer exist
+                    fprintf(1, 'Removed hash for "%s"\n', thisTest);
+                    hashes = rmfield(hashes, thisTest);
+
+                case 'keep'
+                    % useful when not all tests were executed by the tester
+                    fprintf(1, 'Kept hash for "%s"\n', thisTest);
+
+            end
+        end
+    end
+    function action = askActionToPerformOnRemovedTest(testName)
+        % ask which action to carry out on a removed test
+        action = lower(ipp.Results.removedTests);
+        validActions = {'remove', 'keep'};
+        while ~ismember(action, validActions)
+            query = sprintf('Remove or keep "%s"? [Rk]:', testName);
+            answer = strtrim(input(query,'s'));
+
+            if isempty(answer) || strcmpi(answer(1), 'R')
+                action = 'remove';
+            elseif strcmpi(answer(1), 'K')
+                action = 'keep';
+            else
+                action = 'ask';
+            end
+        end
+    end
+    function writeHashesToFile(filename, hashes)
+        % write hashes to a file (or stdout when dry-running)
+        if ~ipp.Results.dryrun
+            fid = fopen(filename, 'w+');
+            finally_fclose_fid = onCleanup(@() fclose(fid));
         else
-            warning('SaveHashTable:NoHashFound',...
-                    'No hash found for "%s". Assuming empty.', S.function);
-            thisHash = ''; % FIXME: when does this happen??
+            fid = 1; % Use stdout to print everything
+            fprintf(fid, '\n\n Output: \n\n');
+        end
+
+        funcNames = sort(fieldnames(hashes));
+        for iFunc = 1:numel(funcNames)
+            func = funcNames{iFunc};
+            fprintf(fid, FILEFORMAT, func, hashes.(func));
         end
-        if ~isempty(thisHash)
-            fprintf(fid, '%s : %s\n', thisFunc, thisHash);
+    end
+    function hashes = readHashesFromFile(filename)
+        % read hashes from a file
+        if exist(filename','file')
+            fid = fopen(filename, 'r');
+            closeFileAfterwards = onCleanup(@() fclose(fid));
```

Comment:
:+1: in general, but you  might want to make the naming consistent, e.g., `finally_fclose_fid`.